tests/test_database_types.py

```diff
@@ -178,60 +242,135 @@ def expand_params(testcase_func, param_num, param):
         self.assertEqual(len(sample_values), self.table2.count())
+        count2_duration = round(time.time() - start, 2)
+
+        # Ensure we actually checksum even for few samples!
+        ch_threshold = 5_000 if N_SAMPLES > DEFAULT_N_SAMPLES else 3
+        ch_factor = (
+            min(max(int(N_SAMPLES / 250_000), 2), 12)
+            if N_SAMPLES > DEFAULT_N_SAMPLES
+            else 2
+        )
+        ch_threads = 12
 
-        differ = TableDiffer(bisection_threshold=3, bisection_factor=2)  # ensure we actually checksum
+        start = time.time()
+        differ = TableDiffer(
+            bisection_threshold=ch_threshold,
+            bisection_factor=ch_factor,
+            max_threadpool_size=ch_threads,
+            debug=True,
+        )
         diff = list(differ.diff_tables(self.table, self.table2))
         expected = []
         self.assertEqual(expected, diff)
         self.assertEqual(0, differ.stats.get("rows_downloaded", 0))
+        checksum_duration = round(time.time() - start, 2)
 
+        start = time.time()
         # Ensure that Python agrees with the checksum!
-        differ = TableDiffer(bisection_threshold=1000000000)
+        dl_factor = int(N_SAMPLES / 100_000) if N_SAMPLES > DEFAULT_N_SAMPLES else 2
+        dl_threshold = int(N_SAMPLES / dl_factor) + 1
+        dl_threads = 8
+
+        differ = TableDiffer(
+            bisection_threshold=dl_threshold,
+            bisection_factor=dl_factor,
+            max_threadpool_size=dl_threads,
+        )
         diff = list(differ.diff_tables(self.table, self.table2))
         expected = []
         self.assertEqual(expected, diff)
         self.assertEqual(len(sample_values), differ.stats.get("rows_downloaded", 0))
+        download_duration = round(time.time() - start, 2)
+
+        print(
+            f"\nsource_db={source_db.__name__} target_db={target_db.__name__} rows={N_SAMPLES} src_table={src_table} target_table={dst_table} source_type={repr(source_type)} target_type={repr(target_type)} insertion_source={insertion_source_duration}s insertion_target={insertion_target_duration}s count_source={count1_duration}s count_target={count2_duration}s checksum={checksum_duration}s download={download_duration}s download_threads={dl_threads} download_bisection_factor={dl_factor} download_bisection_threshold={dl_threshold} checksum_threads={ch_threads} checksum_bisection_factor={ch_factor} checksum_bisection_threshold={ch_threshold}"
+        )
+
+    def _create_table(self, conn, table, type) -> bool:
+        conn.query(f"CREATE TABLE IF NOT EXISTS {table}(id int, col {type})", None)
+        # print(conn.query("SHOW TABLES FROM public", list))
+        # print(conn.query("SELECT * FROM INFORMATION_SCHEMA.TABLES", list))
+        already_seeded = conn.query(f"SELECT COUNT(*) FROM {table}", int) == N_SAMPLES
+        return already_seeded
+
+    def _insert_to_table(self, conn, table, type, values):
+        conn.query(f"TRUNCATE {table}", None)
```

Comment:
Did you make sure TRUNCATE exists in all dialects?

Reply:
It's worked on all I've seen so far. It's fairly standard SQL... I mean until we discover it's not.